src/modules/fundingManager/bondingCurve/formulas/BondingSurface.sol

```diff
@@ -0,0 +1,113 @@
+/// capital available, C_a, and the capital required, C_r. C_a describes the amount of
+/// value stored in the network at any given point in time. C_r  describes the amount
+/// of value that is needed to operate the protocol according to market size
+/// and conditions, the regulatory requirements, as well as the chosen risk appetite,
+/// and allows for considering these three factors in the determination of p
+/// Initial:
+///   B := 10e-7
+///   1 < n <= 2
+///   n := 2
+///
+/// We assume n to be constant. Otherwise we would have to change the burn/mint derivations
+/// on update.
+contract BondingSurface is IBondingSurface, ERC165 {
+    /// @inheritdoc ERC165
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(ERC165)
+        returns (bool)
+    {
+        return interfaceId == type(IBondingSurface).interfaceId
+            || super.supportsInterface(interfaceId);
+    }
+
+    /// @dev constructor
+    constructor() {}
+
+    //--------------------------------------------------------------------------
+    // Public Functions
+
+    /// @inheritdoc IBondingSurface
+    function spotPrice(
+        uint _capitalAvailable,
+        uint _capitalRequirements,
+        uint _basePriceMultiplier
+    ) public pure returns (uint) {
+        uint caSq = FixedPointMathLib.fmul(
+            _capitalAvailable, _capitalAvailable, FixedPointMathLib.WAD
+        ); // C_a^2
+        uint caSqCr = FixedPointMathLib.fdiv(
+            caSq, _capitalRequirements, FixedPointMathLib.WAD
+        );
+        return FixedPointMathLib.fmul(
+            caSqCr, _basePriceMultiplier, FixedPointMathLib.WAD
+        ); // C_a^2 * B / C_r
+    }
+
+    /// @inheritdoc IBondingSurface
+    function tokenOut(
+        uint _in,
+        uint _capitalAvailable,
+        uint _basePriceToCapitalRatio
+    ) public pure returns (uint) {
+        // If the input is bigger inverse will give us 0.
+        if (
+            _capitalAvailable > 1e36 || _capitalAvailable + _in > 1e36
+                || _capitalAvailable == 0
+        ) revert BondingSurface__InvalidInputAmount();
+
+        uint inv1 = _inverse(_capitalAvailable);
+        uint inv2 = _inverse(_capitalAvailable + _in);
+        uint inner = inv1 - inv2;
+
+        return FixedPointMathLib.fmul(
+            _inverse(_basePriceToCapitalRatio), inner, FixedPointMathLib.WAD
+        );
+    }
+
+    /// @inheritdoc IBondingSurface
+    function tokenIn(
+        uint _out,
+        uint _capitalAvailable,
+        uint _basePriceToCapitalRatio
+    ) public pure returns (uint) {
+        // m * (B / C_r)
+        uint BCrM = FixedPointMathLib.fmul(
+            _basePriceToCapitalRatio, _out, FixedPointMathLib.WAD
+        );
+        // 1 / C_a_2
+        uint ca2inv = _inverse(_capitalAvailable);
+
+        return _capitalAvailable - _inverse(BCrM + ca2inv);
```

Comment:
Is this correct? The formula in the docstring is:

```
    ///        B            1
    /// x = (----- * m + -------)^-1
    ///       C_r         C_a_2
    
    ```

Reply:
@rororowyourboat could you please confirm this?

Reply:
The formula implementation has been confirmed by the data department